### DISCUSSION TITLE
ModuleNotFoundError: No module named 'aquacrop.solution.solution_root_zone_water'

### DISCUSSION THREAD

**Discussion by poojarathore** [aquacropos/aquacrop#76] [Q&A] [answered]
Hi all! I am trying to run aqua crop in python but consistently have this error 'ModuleNotFoundError: No module named 'aquacrop.solution.solution_root_zone_water' while calling the package functions after installation.  I am using python version 3.10. Any suggestions?

**Comment by thomasdkelly** [comment, marked as answer]
Hi how have you installed aquacrop?

You should be able to call `python -m aquacrop.scripts.initiate_library ` in the terminal and that should fix this.

**Comment by poojarathore** [comment]
Hi,

I simply 1) created a virtual environment, 2) installed the package aqua
crop in it, 3) launched a Jupiter notebook and tried to run the codes- 'from
 aquacrop import AquaCropModel, Soil, Crop, InitialWaterContent
from aquacrop.utils import prepare_weather, get_filepath'.

I am sorry I am new to python. Earlier I thought there was some path/pip
issue. But it should be resolved by running the package in the virtual env.

On Tue, Sep 13, 2022 at 3:49 PM Tom Kelly ***@***.***> wrote:

> Hi how have you installed aquacrop?
>
> You should be able to call python -m aquacrop.initialize_library in the
> terminal and that should fix this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aquacropos/aquacrop/discussions/76#discussioncomment-3634065>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKIYUEKSDSXKN7FLCCEZMDV6BILPANCNFSM6AAAAAAQLGUDL4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Pooja Rathore
Research Associate| Division of Agricultural Physics|
Indian Agricultural Research Institute| New Delhi, India|
Tel: 011-25844832

**Comment by poojarathore** [comment]
Hi Tom!

I would really appreciate it if you can reply. I haven't been able to
troubleshoot it yet. Don't know if it's a python version or OS (I am using
windows) or any other compatibility issue.

On Tue, Sep 13, 2022 at 5:01 PM Pooja Rathore ***@***.***>
wrote:

> I installed the package using  "pip install aquacrop".
>
> On Tue, Sep 13, 2022 at 4:22 PM Pooja Rathore ***@***.***>
> wrote:
>
>> while running ' python -m aquacrop.initialize_library' in the terminal
>> it is showing this error  - C:\pythonprojects\aquacrop\venv\Scripts\python.exe:
>> No module named aquacrop.initialize_library
>>
>> On Tue, Sep 13, 2022 at 4:18 PM Pooja Rathore ***@***.***>
>> wrote:
>>
>>> Hi,
>>>
>>> I simply 1) created a virtual environment, 2) installed the package aqua
>>> crop in it, 3) launched a Jupiter notebook and tried to run the codes- '
>>> from aquacrop import AquaCropModel, Soil, Crop, InitialWaterContent
>>> from aquacrop.utils import prepare_weather, get_filepath'.
>>>
>>> I am sorry I am new to python. Earlier I thought there was some path/pip
>>> issue. But it should be resolved by running the package in the virtual env.
>>>
>>> On Tue, Sep 13, 2022 at 3:49 PM Tom Kelly ***@***.***>
>>> wrote:
>>>
>>>> Hi how have you installed aquacrop?
>>>>
>>>> You should be able to call python -m aquacrop.initialize_library in
>>>> the terminal and that should fix this.
>>>>
>>>> —
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/aquacropos/aquacrop/discussions/76#discussioncomment-3634065>,
>>>> or unsubscribe
>>>> <https://github.com/notifications/unsubscribe-auth/ADKIYUEKSDSXKN7FLCCEZMDV6BILPANCNFSM6AAAAAAQLGUDL4>
>>>> .
>>>> You are receiving this because you authored the thread.Message ID:
>>>> ***@***.***>
>>>>
>>>
>>>
>>> --
>>> Pooja Rathore
>>> Research Associate| Division of Agricultural Physics|
>>> Indian Agricultural Research Institute| New Delhi, India|
>>> Tel: 011-25844832
>>>
>>>
>>>
>>
>> --
>> Pooja Rathore
>> Research Associate| Division of Agricultural Physics|
>> Indian Agricultural Research Institute| New Delhi, India|
>> Tel: 011-25844832
>>
>>
>>
>
> --
> Pooja Rathore
> Research Associate| Division of Agricultural Physics|
> Indian Agricultural Research Institute| New Delhi, India|
> Tel: 011-25844832
>
>
>

-- 
Pooja Rathore
Research Associate| Division of Agricultural Physics|
Indian Agricultural Research Institute| New Delhi, India|
Tel: 011-25844832

**Comment by thomasdkelly** [comment]
Have you tried running the comand I provided in the anaconda prompt

**Comment by thomasdkelly** [comment]
If you get an error similar to this: `RuntimeError: Attempted to compile AOT function without the compiler used by numpy.distutils present. Cannot find suitable msvc.`, make sure you have installed the msvc compiler for windows (e.g., visual studio). More info here (https://github.com/Ericsson/eva/issues/7). 

I have created a new conda environment on my local windows pc have not run into the same issue

**Comment by poojarathore** [comment]
I installed the package using  "pip install aquacrop".

On Tue, Sep 13, 2022 at 4:22 PM Pooja Rathore ***@***.***>
wrote:

> while running ' python -m aquacrop.initialize_library' in the terminal it
> is showing this error  - C:\pythonprojects\aquacrop\venv\Scripts\python.exe:
> No module named aquacrop.initialize_library
>
> On Tue, Sep 13, 2022 at 4:18 PM Pooja Rathore ***@***.***>
> wrote:
>
>> Hi,
>>
>> I simply 1) created a virtual environment, 2) installed the package aqua
>> crop in it, 3) launched a Jupiter notebook and tried to run the codes- '
>> from aquacrop import AquaCropModel, Soil, Crop, InitialWaterContent
>> from aquacrop.utils import prepare_weather, get_filepath'.
>>
>> I am sorry I am new to python. Earlier I thought there was some path/pip
>> issue. But it should be resolved by running the package in the virtual env.
>>
>> On Tue, Sep 13, 2022 at 3:49 PM Tom Kelly ***@***.***>
>> wrote:
>>
>>> Hi how have you installed aquacrop?
>>>
>>> You should be able to call python -m aquacrop.initialize_library in the
>>> terminal and that should fix this.
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/aquacropos/aquacrop/discussions/76#discussioncomment-3634065>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ADKIYUEKSDSXKN7FLCCEZMDV6BILPANCNFSM6AAAAAAQLGUDL4>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> ***@***.***>
>>>
>>
>>
>> --
>> Pooja Rathore
>> Research Associate| Division of Agricultural Physics|
>> Indian Agricultural Research Institute| New Delhi, India|
>> Tel: 011-25844832
>>
>>
>>
>
> --
> Pooja Rathore
> Research Associate| Division of Agricultural Physics|
> Indian Agricultural Research Institute| New Delhi, India|
> Tel: 011-25844832
>
>
>

-- 
Pooja Rathore
Research Associate| Division of Agricultural Physics|
Indian Agricultural Research Institute| New Delhi, India|
Tel: 011-25844832

**Comment by poojarathore** [comment]
while running ' python -m aquacrop.initialize_library' in the terminal it
is showing this error  - C:\pythonprojects\aquacrop\venv\Scripts\python.exe:
No module named aquacrop.initialize_library

On Tue, Sep 13, 2022 at 4:18 PM Pooja Rathore ***@***.***>
wrote:

> Hi,
>
> I simply 1) created a virtual environment, 2) installed the package aqua
> crop in it, 3) launched a Jupiter notebook and tried to run the codes- '
> from aquacrop import AquaCropModel, Soil, Crop, InitialWaterContent
> from aquacrop.utils import prepare_weather, get_filepath'.
>
> I am sorry I am new to python. Earlier I thought there was some path/pip
> issue. But it should be resolved by running the package in the virtual env.
>
> On Tue, Sep 13, 2022 at 3:49 PM Tom Kelly ***@***.***>
> wrote:
>
>> Hi how have you installed aquacrop?
>>
>> You should be able to call python -m aquacrop.initialize_library in the
>> terminal and that should fix this.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/aquacropos/aquacrop/discussions/76#discussioncomment-3634065>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADKIYUEKSDSXKN7FLCCEZMDV6BILPANCNFSM6AAAAAAQLGUDL4>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Pooja Rathore
> Research Associate| Division of Agricultural Physics|
> Indian Agricultural Research Institute| New Delhi, India|
> Tel: 011-25844832
>
>
>

-- 
Pooja Rathore
Research Associate| Division of Agricultural Physics|
Indian Agricultural Research Institute| New Delhi, India|
Tel: 011-25844832

**Comment by lopezjuana** [comment]
Good evening: I am working on Jupyter Notebook, I have created a python 3.8 working environment and there I installed aquacrop and numpy 1.22. When I execute the commands from aquacrop import AquaCropModel, Soil, Crop, InitialWaterContent
from aquacrop.utils import prepare_weather, get_filepath, I get the error: ModuleNotFoundError: No module named 'aquacrop.solution.solution_root_zone_water'.
I checked and indeed in the solution folder there is no module named solution_root_zone_water, the module is named root_zone_water. Is there any tutorial or step by step guide to install it. I want to use it in my thesis work and I need to know if it will be feasible. Thanks
